src/tools/rbd/Utils.cc

```diff
@@ -815,6 +815,31 @@ int snap_set(librbd::Image &image, const std::string &snap_name) {
   return 0;
 }
 
+bool calc_sparse_extent(const bufferptr &bp, size_t sparse_size, uint64_t length,
+                       size_t &write_offset, size_t &write_length,
```

Comment:
I found the write_offset is not user in this function at all. do we really need this parameter?